cosmovisor/process.go

```diff
@@ -70,12 +73,36 @@ func LaunchProcess(cfg *Config, args []string, stdout, stderr io.Writer) (bool,
 	}
 
 	if upgradeInfo != nil {
+		if err := doBackup(cfg); err != nil {
+			return false, err
+		}
+
 		return true, DoUpgrade(cfg, upgradeInfo)
 	}
 
 	return false, nil
 }
 
+func doBackup(cfg *Config) error {
+	// take backup if `UNSAFE_SKIP_BACKUP` is not set.
+	if !cfg.UnsafeSkipBackup {
+		// a destination directory, Format MM-DD-YYYY
+		dt := time.Now()
+		dst := fmt.Sprintf(cfg.Home+"/data"+"-backup-%s", dt.Format("01-22-2000"))
+
```

Comment:
Would be nice to add a println here saying "Starting to take backup of data directory at time T"